src/Command/SingleDirectoryComponent.php

```diff
@@ -0,0 +1,218 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    private readonly ModuleHandlerInterface $moduleHandler,
+    private readonly ThemeHandlerInterface $themeHandler,
+    private readonly LibraryDiscovery $libraryDiscovery,
+  ) {
+    parent::__construct();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new self(
+      $container->get('module_handler'),
+      $container->get('theme_handler'),
+      $container->get('library.discovery'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function generate(array &$vars, AssetCollection $assets): void {
+    $this->askQuestions($vars);
+    $this->generateAssets($vars, $assets);
+  }
+
+  private function askQuestions(array &$vars): array {
+    $ir = $this->createInterviewer($vars);
+    $vars['machine_name'] = $ir->askMachineName();
+    $vars['name'] = $ir->askName();
+    $vars['directory'] = $ir->ask('Components directory', 'components');
+
+    $vars['component_name'] = $ir->ask('Component name', NULL, new Required());
+    $vars['component_machine_name'] = $ir->ask(
+      'Component machine name',
+      Utils::human2machine($vars['component_name']),
+      new RequiredMachineName(),
+    );
+
+    $vars['component_description'] = $ir->ask(
+      'Component description (optional)',
+    );
+    $choices = [
+      ExtensionLifecycle::STABLE,
+      ExtensionLifecycle::EXPERIMENTAL,
+      ExtensionLifecycle::DEPRECATED,
+      ExtensionLifecycle::OBSOLETE,
+    ];
+    $vars['component_status'] = $ir->choice(
+      'Project type',
+      \array_combine($choices, $choices),
+      ExtensionLifecycle::STABLE,
+    );
+    $vars['component_libraries'] = [];
+
+    do {
+      $library = $this->askLibrary();
+      $vars['component_libraries'][] = $library;
+    } while ($library !== NULL);
+
+    $vars['component_libraries'] = \array_filter($vars['component_libraries']);
+    $vars['component_has_css'] = $ir->confirm('Needs CSS?');
+    $vars['component_has_js'] = $ir->confirm('Needs JS?');
+    if ($ir->confirm('Needs component props?')) {
+      $vars['component_props'] = [];
+      do {
+        $prop = $this->askProp($ir);
+        $vars['component_props'][] = $prop;
+      } while ($ir->confirm('Add another prop?'));
+    }
+    $vars['component_props'] = \array_filter(
+      $vars['component_props'] ?? [],
+    );
+    return $vars;
+  }
+
+  /**
+   * Create the assets that the framework will write to disk later on.
+   *
+   * @param array $vars
+   *   The answers to the CLI questions.
+   * @param \DrupalCodeGenerator\Asset\AssetCollection $assets
+   *   List of all the files to generate.
+   */
+  private function generateAssets(array $vars, AssetCollection $assets): void {
+    if (isset($vars['component_has_css']) && $vars['component_has_css'] !== FALSE) {
```

Comment:
That's alwasy a boolean variable. Isn't it? I suppose `if ($vars['component'])` should be enough.

Reply:
it's enough, but psalm isnt happy about it

Reply:
It needs to be informed about array shape it deals with.
```
@param array{component_has_css: bool, component_has_js: bool} $vars
```